### PR TITLE
Create edit filter

### PR DESCRIPTION
**Pull request by Walid-Sahnoun**

### CODE REVIEW

**Comment by flomillot** [comment]
We should not copy / paste a huge quantity of code like that.
I think an effort have to be done to mutualize the code and put it inside commons-ui.

File: src/components/utils/rhf-inputs/ag-grid-table-rhf/csv-uploader/csv-uploader.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
+/**
```
Comment (YenguiSeddik) [inline src/components/utils/rhf-inputs/ag-grid-table-rhf/csv-uploader/csv-uploader.js:1]: we already have a component to uplode csv ( ImportRuleDialog ). can it be used in this case ?

File: src/components/utils/rhf-inputs/ag-grid-table-rhf/bottom-right-buttons.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+/**
```
Comment (YenguiSeddik) [inline src/components/utils/rhf-inputs/ag-grid-table-rhf/bottom-right-buttons.js:1]: we have a similar component used with dndTable : DndTableBottomRightButtons. can it be used here ?

**Review by YenguiSeddik — CHANGES_REQUESTED** [review]
Some findings after testing

- When editing a filter, this error is show in the dialog: "Name Error while checking name validity: HttpResponseError : 400 Bad Request, message :" even if the name is valid
- When editing, the selection filter dialog is not closed after a filter is selected
- "Explicit naming" is selected by default (in grid explorer "Criteria based" is the default value)
- some missing translation for errors innaming explicit creation like: missingDistributionKeyError

File: src/components/utils/rhf-inputs/ag-grid-table-rhf/bottom-right-buttons.js (hunk 2 above)
Comment (Walid-Sahnoun) [reply src/components/utils/rhf-inputs/ag-grid-table-rhf/bottom-right-buttons.js]: yes, done

File: src/components/dialogs/filter/custom-mui-dialog.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
+/**
```
[the remark this answers is not in the document]
Comment (Walid-Sahnoun) [reply src/components/dialogs/filter/custom-mui-dialog.js]: yes, i used

**Review by FranckLecuyer — APPROVED** [review]
Remark: for a filter with identifier list, when we edit and change the identifiers order, this order is not kept when we reedit the filter ... (another US)

**Review by YenguiSeddik — APPROVED** [review]

File: src/components/dialogs/filter/name-wrapper.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
+/**
```
Comment (Meklo) [inline src/components/dialogs/filter/name-wrapper.js:1]: I used this component for the development of voltage init parameter saving and moved the file to dialogs/commons, do you agree to move it there too so we don't duplicate it ?

**Pull request closed without merging**